[Yee Hao] iP

ou4jairmeiogusrjejiepr opened the pull request:
### Duke Task Organiser🔥🔥🔥

> “Your mind is for having ideas, not holding them.” – David Allen

_**Features**_

- Todo's
- Deadlines
- Events

_**Installation Guide**_

1. Clone this [repo](https://www.github.com/yeehaoo/ip/)
`git clone www.github.com/yeehaoo/ip`
2. Run main method: 
```java
public static void main(String[] args) {
        new Duke("./tasks.txt").run();
}
```

_**Upcoming Features**_

- [ ] GUI support

Commit:
Add Gradle support

ou4jairmeiogusrjejiepr committed:
Level-1

ou4jairmeiogusrjejiepr committed:
Level-2

ou4jairmeiogusrjejiepr committed:
refactored level 2

ou4jairmeiogusrjejiepr committed:
Level-3

ou4jairmeiogusrjejiepr committed:
A-Inheritance

ou4jairmeiogusrjejiepr committed:
A-Inheritance improvements

ou4jairmeiogusrjejiepr committed:
Level4

ou4jairmeiogusrjejiepr committed:
Level-4

ou4jairmeiogusrjejiepr committed:
Level-4

ou4jairmeiogusrjejiepr committed:
Level-5

ou4jairmeiogusrjejiepr committed:
Level-5

ou4jairmeiogusrjejiepr committed:
Level-6

ou4jairmeiogusrjejiepr committed:
Add persistent memory

ou4jairmeiogusrjejiepr committed:
Add ability to parse dates

ou4jairmeiogusrjejiepr committed:
Merge branch 'branch-Level-7' to add persistent memory

ou4jairmeiogusrjejiepr committed:
Merge branch 'branch-Level-8' to add date parsing capabilities

ou4jairmeiogusrjejiepr committed:
Fix bug where constructors for Event and Deadline classes were not being parsed by LocalDate

ou4jairmeiogusrjejiepr committed:
Refactor code to increase conformity to OOP

ou4jairmeiogusrjejiepr committed:
Divide classes into packages, fix a bug where saving to disk did not insert new line after each item

ou4jairmeiogusrjejiepr committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

ou4jairmeiogusrjejiepr committed:
Add gradle support

ou4jairmeiogusrjejiepr committed:
Add 2 JUnit test cases for Event and Todo classes

ou4jairmeiogusrjejiepr committed:
Update gradle settings to configure shadowJar

ou4jairmeiogusrjejiepr committed:
Add Javadocs for all public classes and methods

ou4jairmeiogusrjejiepr committed:
Fix coding standard violations

ou4jairmeiogusrjejiepr committed:
Add ability to search tasks

ou4jairmeiogusrjejiepr committed:
Merge branch 'branch-Level-9'

ou4jairmeiogusrjejiepr committed:
Fix merge conflicts

ou4jairmeiogusrjejiepr committed:
Fix merge conflicts

tlx02 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+    @Override
+    public String toString() {
+        String out = super.getId() + ".[D][";
+        if (super.getStatus()) {
+            out += "X";
+        } else {
+            out += " ";
+        }
+        out += "] " + super.toString() + "(by : " + deadline.format(DateTimeFormatter.ofPattern("MMM dd yyyy")) + ")";
```
Is it possible to rewrite this as String.format so that it is easier to read?

tlx02 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,29 @@
+    @Override
+    public String toString() {
+        String out = super.getId() + ".[E][";
+        if (super.getStatus()) {
```
Would changing the method to isDone() be better, getStatus() might be vague.

tlx02 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,42 @@
+
+    void setNotDone() {
+        isDone = false;
+    }
```
Not a huge issue but perhaps combining these 2 methods can keep your code more clear and concise?
```suggestion
    void setDone(boolean b) {
        isDone = b;
    }
```

tlx02 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,99 @@
+                task = x.substring(9);
+                arrayList.add(new Todo(id, task, done));
+            }
+        });
```
This method seems long and complex, possible to create another method for the lambda expression?

tlx02 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,99 @@
+        System.out.println("Got it. I've added this task: ");
+        System.out.println(todo);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
```
These 3 methods look similar, perhaps you can extract the main idea and put it in a single method: add(Task t).

tlx02 commented on src/main/java/duke/InputParser.java:
```diff
@@ -0,0 +1,50 @@
+                taskList.todo(name);
+            } else {
+                throw new EmptyTextException();
+            }
```
Might want to create new methods for each of these 'else if' blocks to keep your methods short and clear.

tlx02 commented on src/test/java/duke/EventTest.java:
```diff
@@ -0,0 +1,20 @@
+        Task task = new Event(1, "test", "2022-03-05");
+        task.setDone();
+        assertEquals("1.[E][X] test(at : Mar 05 2022)", task.toString());
+    }
```
I love that the tests are clear and straightforward, serving their purpose well.

tlx02 reviewed (commented):
Looks good to merge! Just a few suggestions to increase readability of code.

ou4jairmeiogusrjejiepr committed:
Add GUI capability to Duke

ou4jairmeiogusrjejiepr committed:
Improve GUI capability, chatbot has responses now.

ou4jairmeiogusrjejiepr committed:
Fix bug that prevented parsing of dates

ou4jairmeiogusrjejiepr committed:
Improve readability of code

ou4jairmeiogusrjejiepr committed:
Merge branch 'branch-Level-10'

ou4jairmeiogusrjejiepr committed:
Add assertions to codebase

Currently, codebase does not make use of assertions to verify that some important values were true before the program proceeded. This might lead to unchecked errors.

Assertions should be used to prevent these unchecked errors.

To fix this, assertions were added to verify input at Deadline, Event, InputParser and TaskList classes.

ou4jairmeiogusrjejiepr committed:
Improve code quality

ou4jairmeiogusrjejiepr committed:
Merge pull request #2 from yeehaoo/branch-A-Assertions

Add assertions to codebase

ou4jairmeiogusrjejiepr committed:
Fix merge conflicts arising from A-Assertions

ou4jairmeiogusrjejiepr committed:
Merge pull request #3 from yeehaoo/branch-A-CodeQuality

Improve code quality

ou4jairmeiogusrjejiepr committed:
Fix bug where single word command results in exception

Currently, the command is obtained from user input by looking for the first " " character. However, if the input is a single word command then a -1 is returned, causing string out of bounds error.

To fix this, a check was introduced to prevent substring generation if the input is a single word command.

ou4jairmeiogusrjejiepr committed:
Add Duplicate Detection

Currently, duplicate tasks are added with no checking. This might result in unwanted tasks being added to the tasklist.

To fix this, duplicate checking was added to check for duplicate tasks when adding a new task.

ou4jairmeiogusrjejiepr committed:
Merge branch 'branch-BCD-Extension'

ou4jairmeiogusrjejiepr committed:
Add personality to Duke, now named MOTOONG

ou4jairmeiogusrjejiepr committed:
Add user guide

ou4jairmeiogusrjejiepr committed:
Change gradle main class to Launcher instead of Duke

ou4jairmeiogusrjejiepr committed:
Remove all task IDs

ou4jairmeiogusrjejiepr committed:
Improve and include more javadocs
